openvpn/session/manager.go

```diff
@@ -2,34 +2,39 @@ package session
 
 import (
 	"github.com/mysterium/node/identity"
-	"github.com/mysterium/node/openvpn"
 	"github.com/mysterium/node/session"
 	"sync"
 )
 
+//ServiceConfigProvider interface defines configuration providing dependency
+type ServiceConfigProvider interface {
+	// ProvideServiceConfig function is expected to return service configuration which should be passed to service consumer
+	ProvideServiceConfig() (session.VPNConfig, error)
+}
+
 // NewManager returns session manager which maintains a map of session id -> session
-func NewManager(clientConfigGenerator openvpn.ClientConfigGenerator, idGenerator session.Generator) *manager {
+func NewManager(serviceConfigProvider ServiceConfigProvider, idGenerator session.Generator) *manager {
 	return &manager{
-		idGenerator:          idGenerator,
-		generateClientConfig: clientConfigGenerator,
-		sessionMap:           make(map[session.SessionID]session.Session),
-		creationLock:         sync.Mutex{},
+		idGenerator:    idGenerator,
+		configProvider: serviceConfigProvider,
```

Comment:
SessionManager became not openvpn-specific, lets move out of package `openvpn/session`